### PR TITLE
Provide `exec` module for non-messaging api

### PR DESCRIPTION
**Pull request by peter-cmw**

### CODE REVIEW

File: gstd/src/exec.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// This file is part of Gear.
```
Comment (breathx) [inline gstd/src/exec.rs:1]: Useless file

File: gstd/src/lib.rs [hunk 2]
```diff
@@ -24,6 +24,7 @@
 extern crate galloc;
 
 pub mod bail;
+pub mod exec;
```
Comment (breathx) [inline gstd/src/lib.rs:27]:
```suggestion
pub use gcore::exec;
```

File: gcore/src/msg.rs [hunk 3]
```diff
@@ -21,7 +21,6 @@ use crate::{MessageId, ProgramId};
 
 mod sys {
```
Comment (breathx) [inline gcore/src/msg.rs:22]: Following the idea of this pull request, wouldn't it be reasonable also to move `gr_read()` and `gr_size()` to `gcore/exec` mod?

**Review by breathx — CHANGES_REQUESTED** [review]

**Review by NikVolf — APPROVED** [review]

File: gstd/src/exec.rs (hunk 1 above)
Comment (shamilsan) [reply gstd/src/exec.rs:1]: Why?

Comment (shamilsan) [reply gstd/src/exec.rs:1]: Ah, I see. We can just reimport `gcore::exec` indeed.

File: gcore/src/msg.rs (hunk 3 above)
Comment (shamilsan) [reply gcore/src/msg.rs:22]: In my opinion, both `gr_read` and `gr_size` are related to the message, so there is no reason to move them.

File: gstd/src/exec.rs (hunk 1 above)
Comment (shamilsan) [reply gstd/src/exec.rs]: Fixed.

File: gstd/src/lib.rs (hunk 2 above)
Comment (shamilsan) [reply gstd/src/lib.rs]: Done.

**Review by breathx — APPROVED** [review]

**Pull request merged**